### PR TITLE
Wolf Transformations

### PR DESCRIPTION
**Pull request by Foxwells**
Wolf Pepper is now a TF, and it can give you wolf body parts.

EDIT: aight I got no damn clue why travis is having a hissy, would appreciate it if someone could point me in the right direction. I've been coding for hours, god knows I missed some semicolon somewhere lmao

EDIT2: ffffffuck it it's midnight. I'll look more tomorrow.

### CODE REVIEW

**Comment by Foxwells** [comment]
Would help if the build told me _why_ it failed, thanks--

I'll go prodding for whatever I messed up, gimme a min.

**Comment by Foxwells** [comment]
screenshot_139

is travis having a seizure

File: classes/classes/Appearance.as [hunk 1]
```diff
@@ -2228,7 +2147,7 @@
 					[SKIN_TYPE_GOO, "goo"],
 					[SKIN_TYPE_UNDEFINED, "undefined flesh"],
 					[SKIN_TYPE_DRAGON_SCALES, "scales"],
-					[SKIN_TYPE_FISH_SCALES, "scales"],
+					[SKIN_TYPE_FISH_SCALES, "scales"]
```
Comment (Stadler76) [inline classes/classes/Appearance.as]: Leaving trailing commas in array definitions is not a mistake, but common practice, so adding more items to the end of it only affects one line. That among other advantages.

File: classes/classes/Player.as [hunk 2]
```diff
@@ -786,6 +786,14 @@ use namespace kGAMECLASS;
 				if (isTaur())
 					race = "raccoon-taur";
 			}
+			if (wolfScore() >= 4) {
+				if (player.hasFur()) {
```
Comment (Stadler76) [inline classes/classes/Player.as]: Remove the `player.` from that line.

File: classes/classes/PlayerAppearance.as [hunk 3]
```diff
@@ -138,6 +138,15 @@ package classes
 				if (player.hasScales()) 
 					outputText("  You have the face and head structure of a horse, overlaid with glittering [skinFurScales].");
 			}
+			//wolf-face
+			if (player.faceType == FACE_WOLF) 
+			{
+				outputText("  You have an angular wolf's face complete with a muzzle and black nose");
+				if (player.hasPlainSkin() || player.hasGooSkin() || player.hasScales())
```
Comment (Stadler76) [inline classes/classes/PlayerAppearance.as]: Wouldn't
`if (!player.hasFur())`
make more sense, than excluding all others?

File: classes/classes/Items/Mutations.as [hunk 4]
```diff
@@ -1582,6 +1582,416 @@
...
+    //STAT CHANGES - TOU SPE INT RANDOM CHANCE, LIB LUST COR ALWAYS UPPED
+    dynStats("lib", 1 + rand(2), "lus", 5 + rand(10), "cor", 1 + rand(5));
+    outputText("\n\nYou lick your lips after you finish. That spiciness hit you in more ways than one.", false);
+    if (player.tou < 70 && rand(3) == 0 && changes < changeLimit) {
+        dynStats("tou", (1 * crit));
+        if (crit > 1) outputText("\n\nYou roll your shoulders and tense your arms experimentally. You feel more durable, and your blood seems to run through you more clearly. You know you have more endurance.", false);
+        else outputText("\n\nYour muscles feel denser and more durable. Not so much that feel stronger, but you feel like you can take more hits.", false);
+        changes++;
+    }
+    if (player.spe > 30 && rand(3) == 0 && changes < changeLimit) {
```
Comment (Netys) [inline classes/classes/Items/Mutations.as]: Why? Wolves are not clumsy.

File: classes/classes/Items/Mutations.as [hunk 5]
```diff
@@ -1582,6 +1582,416 @@
...
+        dynStats("spe", (-1 * crit));
+        if (crit > 1) outputText("\n\nThe pepper's strong taste makes you take a couple steps back and lean against the nearest solid object. You don't feel like you'll be moving very fast anymore.", false);
+        else outputText("\n\nYou stumble forward, but manage to catch yourself. Still, though, you feel somewhat slower.", false);
+        changes++;
+    }
+    if (player.inte < 80 && rand(3) == 0 && changes < changeLimit) {
```
Comment (Netys) [inline classes/classes/Items/Mutations.as]: And a feral-sourced TF should not give you such intelligence.

File: classes/classes/Player.as [hunk 6]
```diff
@@ -2537,6 +2572,12 @@ use namespace kGAMECLASS;
 				}
 			}
 			//Alter max stats depending on race
+			if (wolfScore() >= 4) {
+				maxSpe -= 10;
+				maxInt += 10;
```
Comment (Netys) [inline classes/classes/Player.as]: Again, strange to get int and lose spe. I'd suggest to invert it.

File: classes/classes/Items/Mutations.as (hunk 4 above)
Comment (Foxwells) [inline classes/classes/Items/Mutations.as]: I was studying wolf hunting patterns. They are not very speedy, but they are intelligent and have lots of endurance (hence to toughness raise). It wasn't that they're clumsy, they just don't run fast. I will reconsider the stat changes though.

File: classes/classes/PlayerAppearance.as (hunk 3 above)
Comment (Foxwells) [inline classes/classes/PlayerAppearance.as]: Good point cwl I'm a bit new to this so I was referencing somethin' else.

File: classes/classes/Appearance.as (hunk 1 above)
Comment (Foxwells) [inline classes/classes/Appearance.as]: Ah, I see. I saw that there was no comma on one of the Deer additions, so I wondered if that was what was supposed to be. Will adjust :-) Thank you~

**Comment by brrritssocold** [comment]
What are you using locally to build?

Checked out and fixing your branch, so far missing `"` and mistyped types (`number` instead of `Number`).

**Comment by Foxwells** [comment]
Erm, I'm not sure what you mean?

And where are those errors located? I'll get right to fixing them! I'm still learning, so any guidance is appreciated ^^

**Comment by brrritssocold** [comment]
Already fixed, will submit a Pull request to you.

**Comment by Foxwells** [comment]
Of course it was stupid small shit... xD Thank you for spotting those.

**Comment by brrritssocold** [comment]
Please also look at the comments in the [pull request](https://github.com/Foxwells/Corruption-of-Champions-Mod/pull/1), i had to remove one line and edit a variable.
Please confirm the changes are correct.

**Comment by Foxwells** [comment]
We gucci 👌 Thank you so much! ^^ I really want to see Wolf-morphs happen!

**Comment by brrritssocold** [comment]
You are most welcome.

**Comment by Liadri** [comment]
https://docs.google.com/document/d/19uchzhofolix9hCYmt4bWxqMMZzrfeX62LiNyeeKReo/edit?usp=drivesdk

Someone here is late to the party Xianxia had a wolf and even a winter wolf tf one year ago or so. Kitteh just didn't show interest yet into coding it.

**Comment by Foxwells** [comment]
Well, it wasn't added yet, I didn't know about it, and it showed no signs of being made, so. Here we are.

**Pull request merged**